lib/internal/util/comparisons.js

```diff
@@ -489,35 +561,44 @@ function mapEquiv(a, b, strict, memo) {
   return true;
 }
 
-function objEquiv(a, b, strict, keys, memos) {
+function objEquiv(a, b, strict, keys, memos, iterator) {
```

Comment:
Not sure about naming it `iterator` but very much in favour of this change.

Reply:
I am open to suggestions.

Reply:
I'm not sure I like iteratorKind or iterationType are much better to be honest.